velox/exec/Task.h

```diff
@@ -756,20 +785,23 @@ class Task : public std::enable_shared_from_this<Task> {
   // kFinished.
   bool partitionedOutputConsumed_ = false;
 
-  /// Exchange clients. One per pipeline / source.
-  /// Null for pipelines, which don't need it.
-  std::vector<std::shared_ptr<ExchangeClient>> exchangeClients_;
-
-  /// Exchange clients keyed by the corresponding Exchange plan node ID. Used to
-  /// process remaining remote splits after the task has completed early.
-  std::unordered_map<core::PlanNodeId, std::shared_ptr<ExchangeClient>>
-      exchangeClientByPlanNode_;
-
   // Set if terminated by an error. This is the first error reported
   // by any of the instances.
   std::exception_ptr exception_ = nullptr;
   mutable std::mutex mutex_;
 
+  // Exchange clients. One per pipeline / source.
+  // Null for pipelines, which don't need it.
+  //
+  // NOTE: the created exchange clients are also referenced by
+  // 'exchangeClientByPlanNode_'.
+  std::vector<std::shared_ptr<ExchangeClient>> exchangeClients_;
```

Comment:
Maybe it is worth commenting that there can be only one exchange client for a given pipeline ID. Hence, exchange clients can be index either by pipeline ID or by plan node ID.